oeps/oep-0051-bp-conventional-commits.rst

```diff
@@ -0,0 +1,184 @@
+
+This is part of our `Change Transparency`_ initiative.
+
+Motivation
+==========
+
+Developers should describe the changes they make.  This information is valuable to many audiences: deployers, integrators, end-users, operators, and so on. Developers are in the best position to accurately describe their work and its implications.
+
+The nearer to the change that information is captured, the easier it will be to write, and the more accurate it will be.  Changes made by developers are consumed by many people, from the pull request reviewer to the final user of the product.  The more the developer can explain the change, the more easily the change will be understood and used, and the more efficient the overall development process will be.
+
+Commit messages are the closest text to the change itself.  The more
+information-rich we can make commit messages, the more useful they will be.
+
+
+Specification
+============
+
+We are adopting the `Conventional Commits`_ spec, with our own adjustments.
+
+.. note::
+   Many people are familiar with the `Angular commit message format`_, which uses conventional commits and inspired the Conventional Commits spec.  We are **not** adopting the Angular rules.
+
+Git commit messages have these parts::
+
+    <type>: <subject>
+
+    <body>
+
+    <footer>
+
+Type
+----
+
+We use these commit types:
+
+* **build**: anything to do with building or releasing the repo: Makefile, tox.ini, Travis, Jenkins, GitHub actions, and so on.
+
+* **chore**: a repetitive mechanical task such as updating requirements or translations.
+
+* **docs**: documentation-only changes. Documentation changes for other types of work should  be included in those commits. This includes more than the formal docs for a repo, it also covers READMEs, ADRs, docstrings, and so on.
+
+* **feat**: a new feature, or a change to an existing feature. Anything that changes the set of features.  Public API entry points and their behavior are part of the feature set of a product, so changes to those should be marked with "feat".
+
+* **fix**: bug fixes, including changes to the behavior of features, and changes that remove or mitigate security vulnerabilities.
+
+* **perf**: performance improvement.
+
+* **refactor**: code reorganization that doesn't change behavior from the repo consumer perspective.
+
+* **revert**: undo a previous change. The previous commit message, including the type prefix, is included.
+
+* **style**: improve the styling of the code.
+
+* **test**: test-only changes. Adds missing tests or corrects existing tests. Tests accompanying other types of changes go into those commits.
+
+* **exp**: experimental or exploratory changes that are not meant to be permanent.  Occasionally changes have to be pushed to GitHub or even production that are intended to be short-lived. For example, debugging continuous integration, or ad-hoc debugging on live systems.
```

Comment:
I'm not a fan of this; it's opaque to me.
- That the descrption provides two possible _exp_lanations for the _exp_ression :wink: (experimental _or_ exploratory) suggests to me that it's not as obvious as it could be.

I'd prefer `temp` or `tmp`, as these terms have broader adoption across the software landscape, and even beyond the technical domain; this would be clear to less technical readers too.

Reply:
I see your point.  "temp" was my first choice.  I will switch it.